app/routes/thank-you/route.tsx

```diff
@@ -10,9 +9,12 @@ import { OrderSummary } from '~/src/components/order-summary/order-summary';
 import styles from './route.module.scss';
 
 export const loader = async ({ request }: LoaderFunctionArgs) => {
+    const url = new URL(request.url);
+    const orderId = url.searchParams.get('orderId');
     const api = await initializeEcomApiForRequest(request);
-
-    return getThankYouRouteData(api, request.url);
+    // Allow a missing `orderId` to support viewing this page in Codux.
+    const order = orderId ? await api.getOrder(orderId) : undefined;
```

Comment:
Note: inlined the `getThankYouRouteData` loader since there's no business logic to abstract here.

`api.getOrder(orderId)` returns undefined if the order is not found. I think it's better to show the "Thank You" page without the order details then to show the "Order Not Found" page.